main/src/cgeo/geocaching/location/UTMPoint.java

```diff
@@ -0,0 +1,367 @@
+        lat = radToDeg(lat);
+
+        double lon =
+                (d - (1 + 2 * t1 + c1) * d * d * d / 6 + (5 - 2 * c1 + 28 * t1 - 3 * c1 * c1 + 8 * ECC_PRIME_SQUARED + 24 * t1 * t1)
+                        * d * d * d * d * d / 120)
+                        / Math.cos(phi1Rad);
+        lon = longOrigin + radToDeg(lon);
+        return new Geopoint(lat, lon);
+    }
+
+    /**
+     * Converts a set of Longitude and Latitude co-ordinates to UTM.
+     *
+     * @param geopoint the coordinate
+     * @return A UTM class instance containing the value of <code>null</code> if
+     *         conversion failed. If you pass in a UTMPoint, it will be returned
+     *         as well if successful.
+     */
+    public static UTMPoint latLong2UTM(final Geopoint geopoint) {
+        final int zoneNumber = getZoneNumber(geopoint.getLatitude(), geopoint.getLongitude());
+        final double latRad = degToRad(geopoint.getLatitude());
+        final double longRad = degToRad(geopoint.getLongitude());
+
+        // in middle of zone
+        final double longOrigin = (zoneNumber - 1) * 6 - 180 + 3; // +3 puts origin
+        final double longOriginRad = Math.toRadians(longOrigin);
+
+        final double tanLatRad = Math.tan(latRad);
+        final double sinLatRad = Math.sin(latRad);
+        final double cosLatRad = Math.cos(latRad);
+
+        final double n = WGS_84_RADIUS / Math.sqrt(1 - WGS_84_ECC_SQUARED * sinLatRad * sinLatRad);
+        final double t = tanLatRad * tanLatRad;
+        final double c = ECC_PRIME_SQUARED * cosLatRad * cosLatRad;
+        final double a = cosLatRad * (longRad - longOriginRad);
+
+        final double m =
+                WGS_84_RADIUS
+                        * ((1 - WGS_84_ECC_SQUARED / 4 - 3 * ECC_SQUARED_2 / 64 - 5 * ECC_SQUARED_3 / 256) * latRad
+                        - (3 * WGS_84_ECC_SQUARED / 8 + 3 * ECC_SQUARED_2 / 32 + 45 * ECC_SQUARED_3 / 1024) * Math.sin(2 * latRad)
+                        + (15 * ECC_SQUARED_2 / 256 + 45 * ECC_SQUARED_3 / 1024) * Math.sin(4 * latRad) - (35 * ECC_SQUARED_3 / 3072)
+                        * Math.sin(6 * latRad));
+
+        final double utmEasting =
+                (K_0
+                        * n
+                        * (a + (1 - t + c) * a * a * a / 6.0d + (5 - 18 * t + t * t + 72 * c - 58 * ECC_PRIME_SQUARED) * a * a * a
+                        * a * a / 120.0d) + FALSE_EASTING);
+
+        double utmNorthing =
+                (K_0 * (m + n
+                        * Math.tan(latRad)
+                        * (a * a / 2 + (5 - t + 9 * c + 4 * c * c) * a * a * a * a / 24.0d + (61 - 58 * t + t * t + 600 * c - 330 * ECC_PRIME_SQUARED)
+                        * a * a * a * a * a * a / 720.0d)));
+
+        final char zoneLetter = getLetterDesignator(geopoint.getLatitude());
+        if (geopoint.getLatitude() < 0f) {
+            utmNorthing += FALSE_NORTHING;
+        }
+
+        return new UTMPoint(zoneNumber, zoneLetter, utmEasting, utmNorthing);
```

Comment:
If the latest argument was `geopoint.getLatitude() < 0f ? utmNorthing + FALSE_NORTHING : utmNorthing` then `utmNorthing` would not need to be mutable.